exonum-java-binding-common/src/main/java/com/exonum/binding/common/serialization/json/HashCodeJsonSerializer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.serialization.json;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonPrimitive;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+import java.lang.reflect.Type;
+
+/**
+ * {@link HashCode} string serializer. Used to serialize/deserialize HashCode values from/to hex
+ * strings.
+ */
+public final class HashCodeJsonSerializer
+    implements JsonSerializer<HashCode>, JsonDeserializer<HashCode> {
+
+  @Override
+  public JsonElement serialize(HashCode src, Type typeOfSrc, JsonSerializationContext context) {
+    checkNotNull(src, "HashCode value is null");
```

Comment:
What is the difference to failing here or at the next line? As for me, this line brings useless complexity only. 
The behavior will be the same with or without this line.
Also, it'll be great to add a `@throws` section in javadoc

Reply:
I'll update JavaDoc, but prefer to keep checkNotNull()

Reply:
@MakarovS WDYT? 🤔

Reply:
I think it's okay to leave `checkNotNull`, that way we have NPE with some description. Interface method javadoc doesn't say anything, in apidocs example it'd just throw NPE - https://google.github.io/gson/apidocs/com/google/gson/JsonSerializer.html

Reply:
I read parent javadoc a few times and it doesn't say anything about `null`s. So I propose to describe that point. Adding `@throws NullPointerException if src param is null` will be enough.